packages/remix-react/routeModules.ts

```diff
@@ -58,7 +62,7 @@ export type ClientLoaderFunction = ((
  * Arguments passed to a route `clientLoader` function
  */
 export type ClientLoaderFunctionArgs = RRLoaderFunctionArgs<undefined> & {
-  serverLoader: <T = AppData>() => Promise<SerializeFrom<T>>;
+  serverLoader: <T = AppData>() => Promise<SerializeFromImpl<T>>;
```

Comment:
The generics on these calls to the server should only ever be a server loader/action so we can just use `SerializeFromImpl` directly